scripts/BilbyCustomFunctions_Reduction.py

```diff
@@ -0,0 +1,343 @@
+        sys.exit()
+    return array
+
+
+###########################################################################################
+
+def FilesListReduce(filename):
+    """ Creat array of input reduction settings """
+    parameters = []
+    with open(filename) as csv_file:
+        reader = csv.DictReader(csv_file)
+        for row in reader:
+            if row['index'] == '':
+                continue
+            if row['index'] == 'END':
+                break
+            parameters.append(row)
+    return parameters
+
+
+#######################################################################################
+# ===============================================================
+# Function to extract list of lines in the csv file to be processed; input format is a combination of digits, '-' ,',' or empty space
+# returns list of numbers to be processed
+
+def evaluate_files_list(numbers):
+    """ Needed for FilesToReduce, see below """
+
+    expanded = []
+    for number in numbers.split(","):
+        if "-" in number:
+            start, end = number.split("-")
+            nrs = range(int(start), int(end) + 1)
+            expanded.extend(nrs)
+        else:
+            expanded.append(int(number))
+    return expanded
+
+    #######################################################################################
+
+
+def FilesToReduce(parameters, evaluate_files):
+    """ Create list of the files to reduce """
+
+    files_to_reduce = []
+
+    if len(evaluate_files) == 0:
+        files_to_reduce.extend(parameters)
+    else:
+        evaluate_files_l = evaluate_files_list(evaluate_files)  # call funciton for retrieve the IDs list
+        for parameter in parameters:
+            if int(parameter['index']) in evaluate_files_l:
+                files_to_reduce.append(parameter)
+
+    return files_to_reduce
+
+
+#######################################################################################
+
+def strip_NaNs(output_workspace, base_output_name):
+    """  Strip NaNs from the 1D OutputWorkspace """  # add isinf
+
+    data = output_workspace.readY(0)
+    start_index = next((index for index in range(len(data)) if not math.isnan(data[index])), None)
+    end_index = next((index for index in range(len(data) - 1, -1, -1) if not math.isnan(data[index])), None)
+
+    q_values = output_workspace.readX(0)
+    start_q = q_values[start_index]
+    end_q = q_values[end_index]
+
+    CropWorkspace(InputWorkspace=output_workspace, XMin=start_q, XMax=end_q, OutputWorkspace=base_output_name)
+
+    return base_output_name
+
+
+#######################################################################################
+# GENERAL #############################################################################
+#######################################################################################
+
+def GetPixelSize():  # reads current IDF and get pixelsize from there
```

Comment:
Again shouldn't functions in python be in snake_case unless they need to be that due to inheriting from a C++ class or the like?